x/genutil/client/cli/module_hash_by_height.go

```diff
@@ -0,0 +1,107 @@
+package cli
+
+import (
+	"encoding/hex"
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	dbm "github.com/cosmos/cosmos-db"
+	"github.com/spf13/cobra"
+
+	"cosmossdk.io/store/rootmulti"
+	storetypes "cosmossdk.io/store/types"
+
+	"github.com/cosmos/cosmos-sdk/server"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+)
+
+// ModuleHashByHeightQuery retrieves the module hashes at a given height.
+func ModuleHashByHeightQuery[T servertypes.Application](appCreator servertypes.AppCreator[T]) *cobra.Command {
```

Comment:
where is this command being added?

Reply:
I did not add it as I wasn't sure if this was desired as a default and instead have the apps import it themselves, but I can add as a default query for all apps if desired!

Reply:
I moved it to server here to have access to appCreator and registered it as a default CLI cmd https://github.com/cosmos/cosmos-sdk/pull/20779/commits/39ef8218d5ea4b949bf9ca686056cab3fd795a16